lib/api/unversioned/types.go

```diff
@@ -30,8 +30,8 @@ var (
 //
 // All resource and resource lists embed a TypeMetadata as an anonymous field.
 type TypeMetadata struct {
-	Kind       string `json:"kind" validate:"required"`
-	APIVersion string `json:"apiVersion" validate:"required"`
+	Kind       string `json:"kind"`
+	APIVersion string `json:"apiVersion"`
```

Comment:
The Kind and APIVersion are only required for YAML/JSON input.  Since it is a typed interface the kind and version are implied by the structure types - generally, users of the programatic API won't have those fields filled in so validation will always fail - so we need to stop validating them.

calicoctl which does use them performs an explicit check on those fields.